astroid/brain/brain_typing.py

```diff
@@ -165,6 +141,29 @@ def _looks_like_typing_alias(node: nodes.Call) -> bool:
     )
 
 
+def _forbid_class_getitem_access(node: nodes.ClassDef) -> None:
+    """
+    Disable the access to __class_getitem__ method for the node in parameters
+    """
+    if not isinstance(node, nodes.ClassDef):
+        raise TypeError("The parameter type should be ClassDef")
+    try:
+        node.getattr("__class_getitem__")
+        # If we are here, then we are sure to modify object that do have __class_getitem__ method (which origin is one the
+        # protocol defined in collections module) whereas the typing module consider it should not
+        origin_func: node.getattr
+        #  We do not want __class_getitem__ to be found in the classdef
+        def raiser(attr):
+            if attr == "__class_getitem__":
+                raise AttributeInferenceError("__class_getitem__ access is not allowed")
+            else:
+                return origin_func(attr)
+
+        node.getattr = raiser
```

Comment:
```suggestion
        #  We do not want __class_getitem__ to be found in the classdef
        def raiser(origin_func, attr, *args, **kwargs):
            if attr == "__class_getitem__":
                raise AttributeInferenceError("__class_getitem__ access is not allowed")
            else:
                return origin_func(attr, *args, **kwargs)

        node.getattr = partial(raiser, node.getattr)
```

I needed to use `partial` (from `functools`) here to get the acceptance test to pass.

Reply:
Totally agree! I came to the same conclusion. Thanks!